packages/frontend/src/components/canvas/CanvasView.tsx

```diff
@@ -130,9 +131,27 @@ function getDefaultZoom(
   return scale;
 }
 
-const SCALE_FACTOR = 0.2;
+/**
+ * Calculate the position of the mouse relative to the given element
+ **/
+function getRelativeMousePosition(element: HTMLElement, event: MouseEvent) {
+  const rect = element.getBoundingClientRect();
+  return { x: event.clientX - rect.left, y: event.clientY - rect.top };
+}
+
+const SCALE_FACTOR = 0.002;
 const MAX_ZOOM = 100;
-const MIN_ZOOM = 0.5;
+const MIN_ZOOM = 0.9;
+
+const ZOOM_DURATION = 0.1;
+const PAN_DECAY = 0.75;
+// Transition animation on canvas pan and zoom is blurred on Safari and needs to be disabled.
+// If the user spoof their user agent, this is not my problem.
+// Bug in question https://bugs.webkit.org/show_bug.cgi?id=27684
+const IS_SAFARI =
+  navigator.userAgent.includes("Safari/") &&
+  !navigator.userAgent.includes("Chrome/") &&
+  !navigator.userAgent.includes("Chromium/");
```

Comment:
This doesn't work on the server, so we need to add a check.

```suggestion
const IS_SERVER = typeof window === "undefined";

const ZOOM_DURATION = 0.1;
const PAN_DECAY = 0.75;
// Transition animation on canvas pan and zoom is blurred on Safari and needs to be disabled.
// If the user spoof their user agent, this is not my problem.
// Bug in question https://bugs.webkit.org/show_bug.cgi?id=27684
const IS_SAFARI =
  !IS_SERVER &&
```

Reply:
Seems like that could be an `isServer()` util function?

In any case, I would _also_ extract `isSafari` into its own helper

Reply:
Does this still run on the server when `"use client"` has been defined? Not sure if `IS_SAFARI` is being used outside of this file yet.